api/database/seeders/JobPosterTemplateSeeder.php

```diff
@@ -0,0 +1,122 @@
+
+        // Check for duplicate reference ids in templates
+        $referenceIds = array_map(
+            function ($model) {
+                return $model->referenceId;
+            },
+            $templatesModels
+        );
+        if (count(array_unique($referenceIds)) != count($templatesModels)) {
+            throw new Exception('The reference ids are not unique');
+        }
+
+        // TEMPLATES
+        // update or create all the templates
+        foreach ($templatesModels as $templateModel) {
+            $classificationObject = DB::table('classifications')
+                ->where('group', $templateModel->classification->group)
+                ->where('level', $templateModel->classification->level)
+                ->first();
+
+            JobPosterTemplate::updateOrCreate(
+                ['reference_id' => $templateModel->referenceId],
+                [
+                    'stream' => $templateModel->stream->value,
+                    'classification_id' => $classificationObject->id,
+                    'supervisory_status' => $templateModel->supervisoryStatus,
+                    'name' => [
+                        'en' => $templateModel->name->en,
+                        'fr' => $templateModel->name->fr,
+                    ],
+                    'description' => [
+                        'en' => $templateModel->description->en,
+                        'fr' => $templateModel->description->fr,
+                    ],
+                    'work_description' => [
+                        'en' => $templateModel->workDescription->en,
+                        'fr' => $templateModel->workDescription->fr,
+                    ],
+                    'tasks' => [
+                        'en' => $templateModel->tasks->en,
+                        'fr' => $templateModel->tasks->fr,
+                    ],
+                    'keywords' => [
+                        'en' => $templateModel->keywords->en,
+                        'fr' => $templateModel->keywords->fr,
+                    ],
+                    'essential_technical_skills_notes' => [
+                        'en' => $templateModel->essentialTechnicalSkillsNotes->en,
+                        'fr' => $templateModel->essentialTechnicalSkillsNotes->fr,
+                    ],
+                    'essential_behavioural_skills_notes' => [
+                        'en' => $templateModel->essentialBehaviouralSkillsNotes->en,
+                        'fr' => $templateModel->essentialBehaviouralSkillsNotes->fr,
+                    ],
+                    'nonessential_technical_skills_notes' => [
+                        'en' => $templateModel->nonessentialTechnicalSkillsNotes->en,
+                        'fr' => $templateModel->nonessentialTechnicalSkillsNotes->fr,
+                    ],
+                ]
+            );
+        }
+
+        // SKILLS
+        // wipe existing pivot entries
+        DB::table('job_poster_template_skill')->truncate();
```

Comment:
Destructive as far as skills concerned unfortunately

Reply:
This way I can ensure there are no unwanted skills around from re-runs, could possibly be improved  🤔

Reply:
Yeah, I definitely don't love having a truncate in a seeder.  Let's try avoid this somehow.

Reply:
If you check api/database/seeders/SkillSeeder.php you can see how the relationship `sync` is used rather than using DB:table directly.  Could you try something like that instead?

Reply:
Okay, I brought back something I was trying earlier in this PR. How does this look? https://github.com/GCTC-NTGC/gc-digital-talent/pull/11523/commits/f98e5785a2f1865e2783a37c00f018ac192c6af4

Reply:
Yeah, I like the looks of that much better.  A couple of quick comments on it:
1) On L177 you can probably start your query on the Skill model instead of DB::table.
2) On L169 try getting a list of skill IDs instead of keys that you need to look up models with later. Because...
3) On L180, if you build a collection of IDs first you can feed them in one-shot using `sync` instead of `syncWithoutDetaching` which avoids the need to do a cleanup of unused IDs from L188 - L196, too.

Let me know if you want to talk through these suggestions.  :+1:

Reply:
> On L177 you can probably start your query on the Skill model instead of DB::table.

I was trying to optimize performance a bit by using the facade for skill fetching, maybe needed maybe not

> On L169 try getting a list of skill IDs instead of keys that you need to look up models with later. Because...

That won't work in this case as I am pulling from JSON in that line and the key field is what is constant across different environments. I need the keys then to grab appropriate id values

>On L180, if you build a collection of IDs first you can feed them in one-shot using sync instead of syncWithoutDetaching which avoids the need to do a cleanup of unused IDs from L188 - L196, too.

Took a while to figure out, but got something going here https://github.com/GCTC-NTGC/gc-digital-talent/pull/11523/commits/07c261906cf55c581cfbf7f43f79119e7324be89
Got rid of the some things with it like the array of key building and clean-up

Reply:
Nice, this looks great.  I'd suggest using `sole` on your skill lookup in case there's a bad skill key but otherwise, I love it.

Reply:
https://github.com/GCTC-NTGC/gc-digital-talent/pull/11523/commits/62a58fe3b25de3ffbb57f1a1a27132f32241e42d